### ISSUE TITLE
mrc_multiplexer.js errors out when URL is pasted in MRC chat

### ISSUE THREAD

**Issue by voipmeister** [NuSkooler/enigma-bbs#551]
**Describe the Bug**
Enigma crashes when a URL is pasted in the MRC chat.

**To Reproduce**
Log onto MRC from Enigma 0.0.14, have someone paste for instance: `https://twitch.tv/the_meatlotion`, from a Mystic board (probably any UTF-8 client). In this case it was Meatlotion.

**Expected Behavior**
Proper sanitation, no crash. Other chatters had no issue in the same chat, but they use different clients obviously.

**Actual Behavior**
Enigma crashes:
```
ENiGMA½ Copyright (c) 2015-2024, Bryan D. Ashby
_____________________   _____  ____________________    __________\_   /
/____       _____|      __________       ___|__|      ____|     \   /  _____  \
---- \______\ -- |______\ ------ /______/ ---- |______\ - |______\ /__/ // ___/
                                                                     /__   _\
 <*>   ENiGMA½  // HTTPS://GITHUB.COM/NUSKOOLER/ENIGMA-BBS   <*>       /__/

-------------------------------------------------------------------------------

System started!
/home/enigma/enigma-bbs/core/servers/chat/mrc_multiplexer.js:249
                message.to_user.toUpperCase() == client.username.toUpperCase() ||
                                ^

TypeError: Cannot read properties of undefined (reading 'toUpperCase')
    at /home/enigma/enigma-bbs/core/servers/chat/mrc_multiplexer.js:249:33
    at Set.forEach (<anonymous>)
    at MrcModule.sendToClient (/home/enigma/enigma-bbs/core/servers/chat/mrc_multiplexer.js:245:26)
    at MrcModule.receiveFromMRC (/home/enigma/enigma-bbs/core/servers/chat/mrc_multiplexer.js:323:18)
    at /home/enigma/enigma-bbs/core/servers/chat/mrc_multiplexer.js:157:30
    at Array.forEach (<anonymous>)
    at handleData (/home/enigma/enigma-bbs/core/servers/chat/mrc_multiplexer.js:153:19)
    at TLSSocket.<anonymous> (/home/enigma/enigma-bbs/core/servers/chat/mrc_multiplexer.js:164:13)
    at TLSSocket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:368:12)

Node.js v18.20.4
[enigma@node01 enigma-bbs]$
```

**Screenshots**
N/a.

**Environment**
* [x] I am using Node.js v14.x LTS or higher
* [x] `npm install` or `yarn` reports success
* Actual Node.js version (`node --version`): `v18.20.4`
* Operating system (`uname -a` on *nix systems): `Linux node01 4.18.0-553.16.1.el8_10.x86_64 #1 SMP Thu Aug 8 17:47:08 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
* Revision (`git rev-parse --short HEAD`): `9c0bc88f`
* Any additional relevant information about your setup: Basically unmodded, one FTN connected.

**Comment by voipmeister** [comment]
ML and I have been testing some more, and the above happens when the URL shortener kicks in. It 'works' both ways:
* when ML sends a link in a channel I'm connected to via Enigma 0.0.14, Enigma crashes
* when I send a link in a channel, the same happens

**Comment by voipmeister** [comment]
Testing a bit more, when commenting the middle line below (row 249 in `mrc_multiplexer.js`), the problem is gone. However, it prevents private messages from being sent.

```
                // Fix PrivMSG delivery on case mismatch
                //message.to_user.toUpperCase() == client.username.toUpperCase() ||
                message.to_user == 'CLIENT' ||
```

**Comment by stack-fault** [comment]
This issue is resolved with the PR merged

**Comment by voipmeister** [comment]
Closed, as resolved in #552
